pco/src/latent_batch_dissector.rs

```diff
@@ -42,16 +49,33 @@ impl<'a, L: Latent> LatentBatchDissector<'a, L> {
       }
     }
 
-    let max_idx = self.table.infos.len().saturating_sub(1);
-    search_idxs
-      .iter_mut()
-      .for_each(|search_idx| *search_idx = min(*search_idx, max_idx));
+    let n_bins = self.table.infos.len();
+    if n_bins < 1 << self.table.search_size_log {
+      // We worked with a balanced binary tree with missing leaves filled, so it
+      // might have overshot some bin indices.
```

Comment:
Was this condition previously only hit in the case where `self.table.infos.len()` is either 0 or 1?  From this description it sounds like it could have happened more generally?